snap/info_snap_yaml_test.go

```diff
@@ -2176,20 +2176,88 @@ components:
 `))
 	c.Assert(err, IsNil)
 	c.Check(info.InstanceName(), Equals, "snap")
-	c.Check(info.Components, DeepEquals, map[string]snap.Component{
+	c.Check(info.Components, DeepEquals, map[string]*snap.Component{
 		"test1": {
 			Type:        "test",
 			Summary:     "test component",
 			Description: "long component description",
+			Name:        "test1",
 		},
 		"test2": {
 			Type:        "test",
 			Summary:     "test component 2",
 			Description: "long component description 2",
+			Name:        "test2",
 		},
 	})
 }
 
+func (s *YamlSuite) TestUnmarshalComponentsHooks(c *C) {
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+components:
+  test1:
+    type: test
+    summary: test component
+    description: long component description
+    hooks:
+      install:
+        plugs: [network, network-bind]
+      pre-refresh:
+        plugs: [test-plug]
+      post-refresh:
+      remove:
```

Comment:
we need tests with multiple component, same hook and mix of same and different plugs